core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/aggregate/AggregationOperation.scala

```diff
@@ -10,8 +10,38 @@ import edu.uci.ics.texera.workflow.common.tuple.schema.{Attribute, AttributeType
 
 import java.sql.Timestamp
 
+@JsonSchemaInject(json =
+  """
+{
+  "attributeType1": {
+    "attribute": {
+      "allOf": [
+        {
+          "if": {
+            "aggFunction": {
+              "enum": ["sum", "average", "min", "max"]
+            }
+          },
+          "then": {
+            "enum": ["integer", "long", "float", "double"]
+          }
+        },
+        {
+          "if": {
+            "aggFunction": {
+              "enum": ["concat"]
+            }
+          },
+          "then": {
+            "enum": ["string"]
+          }
+        }
+      ]
+    }
+  }
+}
+""")
 class AggregationOperation() {
```

Comment:
Should this annotation be injected to the operation or to the aggregate desc?

Reply:
Discussed offline:

I injected it there so that I don't need to deal with references. The field that needs type validation is in properties/items/properties/attribute instead of properties/attribute . If I injected it in the outside operator schema, I would need to refer to attribute as items/attribute, and add code in front-end to handle this case.